tests/tests_full_integration_dyndns.yml

```diff
@@ -0,0 +1,186 @@
+
+- name: Configure DynDNS on AD
+  hosts: ad
+  gather_facts: true
+  tasks:
+    - name: Get AD address
+      set_fact:
+        ad_address: "{{ hostvars[inventory_hostname].ansible_host }}"
+    - name: Get network for zone
+      set_fact:
+        network_ad: >-
+        {{ ad_address.split('.')[-2::-1] | join('.') }}.in-addr.arpa
+    - name: Show AD network
+      debug:
+        msg: "AD network: {{ network_ad }}"
+    - name: Create reverse zone
+      win_command: dnscmd.exe /zoneadd {{ network_ad }} /primary
+      register: res
+      failed_when:
+        - 'res.rc != 0 and "DNS_ERROR_ZONE_ALREADY_EXISTS" not in res.stdout'
+    - name: Allow updates in the zone
+      win_command: >-
+        dnscmd.exe /config {{ hostvars[groups['client'][0]].ad_integration_realm }}
+         /allowupdate 1
+    - name: Allow updates in the reverse zone
+      win_command: dnscmd.exe /config {{ network_ad }} /allowupdate 1
+    - name: Disable dns forwarders
+      win_command: dnscmd.exe /config /norecursion 1
+
+- name: Ensure that the role configures dynamic dns
+  hosts: client
+  tasks:
+    - name: Run the integration
+      when: "'ad' in groups and groups['ad']"
+      block:
+        - name: Debug dyndns settings
+          debug:
+            msg: >-
+              Interface {{ _ad_dyndns_iface | default('eth0') }} and server
+              : {{ hostvars[groups['ad'][0]].ansible_host }}
+        - name: Run the system role with proper config
+          include_role:
+            name: linux-system-roles.ad_integration
+          vars:
+            ad_dyndns_server: "{{ hostvars[groups['ad'][0]].ansible_host }}"
+            ad_dyndns_iface: "{{ _ad_dyndns_iface | default('eth0') }}"
+            ad_dyndns_auth: "none"
+            ad_dyndns_update: true
+            ad_dyndns_refresh_interval: 60
+        - name: Apply additional changes on sssd
+          ini_file:
+            path: /etc/sssd/sssd.conf
+            state: present
+            section: "domain/{{ ad_integration_realm | lower }}"
+            option: "{{ item.key }}"
+            value: "{{ item.value }}"
+            create: true
+            owner: root
+            group: root
+            mode: 0600
+          loop:
+            - key: debug_level
+              value: 9
+        - name: Clean sssd log
+          command: >-
+            truncate -s 0  /var/log/sssd/sssd_{{ ad_integration_realm }}.log
+        - name: Restart sssd
+          service:
+            name: sssd
+            status: restarted
+        - name: Pause for 5 to give sssd chance to refresh dn record on AD
+          ansible.builtin.pause:
+            minutes: 5
+        - name: Check sssd log fo dydndns update info
+          command: >-
+            grep -A 20 "nsupdate"  /var/log/sssd/sssd_{{ ad_integration_realm }}.log
+        - name: Grab sssd.conf contents
+          command: cat /etc/sssd/sssd.conf
+          register: command_output
+        - name: Print sssd.conf
```

Comment:
You don't need this task - `command_output` isn't used anywhere else, and using `command: cat ...` without a `register` will display the file contents to stdout in the log